File: data/snippets/quiz006.cpp
class ReportGenerator {
public:
    void generateReport(
        std::string reportTitle, 
        std::string reportDate, 
        std::string companyName, 
        std::string departmentName,
        std::string authorName, 
        std::string authorEmail, 
        std::string authorPhone,
        int pageCount, 
        bool includeCharts, 
        bool includeExecutiveSummary, 
        bool includeTables, 
        std::string outputFormat, 
        std::string outputPath, 
        std::string colorTheme) 
    {
        std::cout << "Generating report: " << reportTitle << std::endl;
        std::cout << "Report date: " << reportDate << std::endl;
        std::cout << "For: " << companyName << ", " << departmentName << std::endl;
        std::cout << "Author: " << authorName << " (" << authorEmail << ", " << authorPhone << ")" << std::endl;
        std::cout << "Pages: " << pageCount << std::endl;
        
        std::cout << "Including:";
        if (includeCharts) std::cout << " Charts";
        if (includeExecutiveSummary) std::cout << " Executive Summary";
        if (includeTables) std::cout << " Tables";
        std::cout << std::endl;
        
        std::cout << "Output format: " << outputFormat << std::endl;
        std::cout << "Output path: " << outputPath << std::endl;
        std::cout << "Theme: " << colorTheme << std::endl;
        
        // 生成报告的具体逻辑...
    }
};

File: data/snippets/quiz006.java
public class ReportGenerator {
    
    public void generateReport(String reportTitle, String reportDate, String companyName, 
                             String departmentName, String authorName, String authorEmail, 
                             String authorPhone, int pageCount, boolean includeCharts, 
                             boolean includeExecutiveSummary, boolean includeTables, 
                             boolean includeFootnotes, String outputFormat, String outputPath, 
                             String colorTheme, String fontFamily, int fontSize, 
                             boolean enableWatermark, String watermarkText) {
        
        System.out.println("Generating report: " + reportTitle);
        System.out.println("Report date: " + reportDate);
        System.out.println("For: " + companyName + ", " + departmentName);
        System.out.println("Author: " + authorName + " (" + authorEmail + ", " + authorPhone + ")");
        System.out.println("Pages: " + pageCount);
        
        System.out.print("Including:");
        if (includeCharts) System.out.print(" Charts");
        if (includeExecutiveSummary) System.out.print(" Executive Summary");
        if (includeTables) System.out.print(" Tables");
        if (includeFootnotes) System.out.print(" Footnotes");
        System.out.println();
        
        System.out.println("Output format: " + outputFormat);
        System.out.println("Output path: " + outputPath);
        System.out.println("Theme: " + colorTheme);
        System.out.println("Font: " + fontFamily + ", size " + fontSize);
        
        if (enableWatermark) {
            System.out.println("Watermark: " + watermarkText);
        }
        
        // create report ....
    }
}
